// johnny-five/lib/motor-shields.js
module.exports = {
   // .. this can be shield specific stuff, I'll use adafruit v1 as an example:
  ADAFRUIT_V1: {
    M1: {
      pwm: 11,
      register: { data: 8, clock: 4, latch: 12 },
      bits: { a: 2, b: 3 }
    },
    M2: {
      pwm: 3,
      register: { data: 8, clock: 4, latch: 12 },
      bits: { a: 1, b: 4 }
    },
    M3: {
      pwm: 6,
      register: { data: 8, clock: 4, latch: 12 },
      bits: { a: 5, b: 7 }
    },
    M4: {
      pwm: 5,
      register: { data: 8, clock: 4, latch: 12 },
      bits: { a: 0, b: 6 }
    }
  },
  SEEED_STUDIO: {
    A: {
      pins: {
        pwm: 9,
        dir: 8,
        cdir: 11
      }
    },
    B: {
      pins: {
        pwm: 10,
        dir: 12,
        cdir: 13
      }
    }
  },
  FREETRONICS_HBRIDGE: {
    A: {
      pins: {
        pwm: 6,
        dir: 4,
        cdir: 7
      }
    },
    B: {
      pins: {
        pwm: 5,
        dir: 3,
        cdir: 2
      }
    }
  },
  ARDUINO_MOTOR_SHIELD_R3_1: {
    A: {
      pins: {
        pwm: 3,
        dir: 12,
      }
    },
    B: {
      pins: {
        pwm: 11,
        dir: 13,
      }
    }
  },
  ARDUINO_MOTOR_SHIELD_R3_2: {
    A: {
      pins: {
        pwm: 3,
        dir: 12,
        brake: 9
      }
    },
    B: {
      pins: {
        pwm: 11,
        dir: 13,
        brake: 8
      }
    }
  },
  ARDUINO_MOTOR_SHIELD_R3_3: {
    A: {
      pins: {
        pwm: 3,
        dir: 12,
        brake: 9,
        current: "A0"
      }
    },
    B: {
      pins: {
        pwm: 11,
        dir: 13,
        brake: 8,
        current: "A1"
      }
    }
  }
};
